go/vt/vtgate/semantics/info_schema.go

```diff
@@ -0,0 +1,973 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semantics
+
+import (
+	"sort"
+	"strings"
+
+	"vitess.io/vitess/go/mysql/collations"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/proto/query"
+	topodatapb "vitess.io/vitess/go/vt/proto/topodata"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+)
+
+// getInfoSchema returns a map of all information_schema tables and their columns with types
+// To recreate this information from MySQL, you can run the test in info_schema_gen_test.go0
+func getInfoSchema() map[string]map[string]query.Type {
+	infSchema := map[string]map[string]query.Type{}
```

Comment:
nit: maybe adding a link to the doc could help to know what each table/column is used for: https://dev.mysql.com/doc/refman/8.0/en/information-schema-table-reference.html